libtsuba/src/RDGSlice.cpp

```diff
@@ -11,6 +11,142 @@
 #include "tsuba/RDGPrefix.h"
 #include "tsuba/RDGTopology.h"
 
+namespace {
+katana::Result<void>
+demand_load_metadata_array(
+    std::string array_name, std::function<void()> empty, tsuba::RDGCore* core) {
+  auto it = std::find_if(
+      core->part_header().part_prop_info_list().begin(),
+      core->part_header().part_prop_info_list().end(),
+      [&array_name](tsuba::PropStorageInfo& psi) {
+        return psi.name() == array_name;
+      });
+
+  if (it == core->part_header().part_prop_info_list().end()) {
+    // caller's function for emptying the array we care about
+    empty();
```

Comment:
I think this should be documented a bit more to explain why the empty function needs to be passed in. I see that this function is used internally mainly, but still worth a bit more explanation since the burden is placed on the caller.

Reply:
I added some documentation, is it more clear now?

Reply:
Yes it is much clearer now.